Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift

```diff
@@ -0,0 +1,136 @@
+                                                         isDestructive: false)
+
+    // MARK: - Input
+
+    public let syncButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let leftNavBarButtonDidTap = PassthroughRelay<Void>()
+
+    // MARK: - Private
+
+    private let interactor: CourseSyncSelectorInteractor
+    private var subscriptions = Set<AnyCancellable>()
+
+    init(interactor: CourseSyncSelectorInteractor) {
+        self.interactor = interactor
+
+        updateState(interactor)
+        updateSyncButtonState(interactor)
+        updateConfirmationDialogMessage(interactor)
+        updateSelectAllButtonTitle(interactor)
+
+        handleLeftNavBarTap(interactor)
+        handleSyncButtonTap(interactor, confirmAlert: confirmAlert)
+    }
+
+    private func updateSyncButtonState(_ interactor: CourseSyncSelectorInteractor) {
+        interactor
+            .observeSelectedCount()
+            .map { $0 == 0 }
+            .assign(to: &$syncButtonDisabled)
+    }
+
+    private func updateSelectAllButtonTitle(_ interactor: CourseSyncSelectorInteractor) {
+        interactor
+            .observeIsEverythingSelected()
+            .map { $0 ? NSLocalizedString("Deselect All", comment: "")
+                      : NSLocalizedString("Select All", comment: "") }
+            .assign(to: &$leftNavBarTitle)
+    }
+
+    private func updateConfirmationDialogMessage(_ interactor: CourseSyncSelectorInteractor) {
+        interactor
+            .observeSelectedCount()
+            .map { itemCount in
+                let format = NSLocalizedString("There are %d items selected for offline availability. The selected content will be downloaded to the device.", bundle: .core, comment: "")
+                return String.localizedStringWithFormat(format, itemCount)
+            }
+            .assign(to: \.message, on: confirmAlert, ownership: .weak)
+            .store(in: &subscriptions)
+    }
+
+    private func handleLeftNavBarTap(_ interactor: CourseSyncSelectorInteractor) {
+        leftNavBarButtonDidTap
+            .flatMap { interactor.observeIsEverythingSelected().first() }
+            .toggle()
+            .sink { interactor.toggleAllCoursesSelection(isSelected: $0) }
+            .store(in: &subscriptions)
+    }
+
+    private func handleSyncButtonTap(_ interactor: CourseSyncSelectorInteractor,
+                                     confirmAlert: ConfirmationAlertViewModel) {
+        syncButtonDidTap
+            .handleEvents(receiveOutput: { [unowned self] _ in
+                isShowingConfirmationDialog = true
+            })
+            .flatMap { view in
+                confirmAlert.userConfirmation().map { view }
+            }
+            .flatMap { view in
+                interactor.getSelectedCourseEntries()
+                    .handleEvents(receiveOutput: { _ in
+                        // TODO: Start download, go to dashboard
+                        AppEnvironment.shared.router.dismiss(view)
```

Comment:
If we move the `AppEnvironment` from the view to this view model, we could reference it here as well.

Reply:
Added to the follow-up ticket.